developer/src/kmcmplib/src/Compiler.cpp

```diff
@@ -3863,17 +3863,68 @@ bool hasPreamble(std::u16string result) {
   return result.size() > 0 && result[0] == 0xFEFF;
 }
 
+bool isValidUtf8(KMX_BYTE* str, int sz) {
```

Comment:
This fairly complex function seems to be very lightly tested. Is it original code or copied from somewhere?